alacritty_terminal/src/term/mod.rs

```diff
@@ -422,11 +423,11 @@ impl<'a, C> Iterator for RenderableCellsIter<'a, C> {
                     let mut cell = RenderableCell::new(self, cell);
                     cell.inner = RenderableCellContent::Cursor(self.cursor.key);
 
-                    // Only apply static color if it isn't close to the cell's current background.
-                    if let CellRgb::Rgb(color) = self.cursor.cursor_color {
-                        if color.contrast(cell.bg) >= MIN_CURSOR_CONTRAST {
-                            cell.fg = self.cursor.cursor_color.color(cell.fg, cell.bg);
-                        }
+                    // Only apply static color if it isn't close to the cell's current
+                    // background or refers to cell colors.
+                    match self.cursor.cursor_color {
+                        CellRgb::Rgb(color) if color.contrast(cell.bg) < MIN_CURSOR_CONTRAST => (),
+                        _ => cell.fg = self.cursor.cursor_color.color(cell.fg, cell.bg),
```

Comment:
I think the following would be better, it requires 1.42.0 though. But chances are that's fine for 0.6.0?

```suggestion
                    if !matches(self.cursor.cursor_color, CellRgb::Rgb(color) if color.contrast(cell.bg) < MIN_CURSOR_CONTRAST) {
                        cell.fg = self.cursor.cursor_color.color(cell.fg, cell.bg);
```

Reply:
I'd bump straight to `v0.4.3` if anything in a separate PR, are you fine with that?

Reply:
0.4.3?

Reply:
In general I'm opposed to bumping minimum Rust version separately, since that makes it harder to track down the original reason for the bump.

Reply:
>0.4.3?

Yeah, I want `0.4.3` for cfg before if.

Reply:
You mean 1.43.0?

Reply:
Also I've looked at the other PR and if you want to bump it for multiple different reasons anyways, I'd recommend merging this *without* the `matches!` macro and then fixing it immediately in the rust version bump PR.

Reply:
ok.